client/state/login/reducer.js

```diff
@@ -174,29 +174,29 @@ export const socialAccount = createReducer( { isCreating: false, createError: nu
 		bearerToken,
 		createError: null,
 	} ),
-	[ SOCIAL_LOGIN_REQUEST_FAILURE ]: ( state, { error, authInfo } ) => ( {
+	[ SOCIAL_LOGIN_REQUEST_FAILURE ]: ( state, { error } ) => ( {
 		...state,
 		requestError: error,
-		email: error.email,
-		authInfo
 	} ),
 	[ USER_RECEIVE ]: state => ( { ...state, bearerToken: null, username: null, createError: null, } ),
 	[ LOGIN_REQUEST ]: state => ( { ...state, createError: null } ),
 } );
 
-export const socialAccountLink = createReducer( { isLinking: false }, {
-	[ SOCIAL_CREATE_ACCOUNT_REQUEST_FAILURE ]: ( state, { error, token, service } ) => {
-		if ( error.code === 'user_exists' ) {
-			return {
-				isLinking: true,
-				email: error.email,
-				token,
-				service,
-			};
-		}
+const userExistsErrorHandler = ( state, { error, authInfo } ) => {
```

Comment:
I find a bit confusing that we have to pass `authInfo` for both `SOCIAL_CREATE_ACCOUNT_REQUEST_FAILURE ` and `SOCIAL_LOGIN_REQUEST_FAILURE`. Couldn't we just get it from the state? (of course `SOCIAL_CREATE_ACCOUNT_REQUEST` and `SOCIAL_LOGIN_REQUEST` would need to store it there first).